### PR TITLE
Reference form-flow-library-yaml

### PR DESCRIPTION
**Pull request by bencalegari**
- Consumers now need to use form-flow-library for test rather than form-flow-library test.. should be the only change they'll need to make!

[#184056302]

### CODE REVIEW

File: .github/workflows/run-tests.yaml [hunk 1]
```diff
@@ -13,6 +13,16 @@ jobs:
   test:
     name: Run unit tests
     runs-on: ubuntu-latest
+    services:
```
Comment (coltborg) [inline .github/workflows/run-tests.yaml:16]: To get the tests to pass in this PR, I had to apply the [same code as we did in the library `run_tests`](https://github.com/codeforamerica/form-flow/blob/6b54e1454f995c8198abdd74b9cefb08c59f7e80/.github/workflows/run-tests.yml#L12).

This doesn't seem ideal, or is this the only way to get tests to work?

File: src/test/resources/application-test.yaml [hunk 2]
```diff
@@ -12,9 +12,14 @@ form-flow:
       auth-token: 'test-token'
 spring:
   datasource:
-    url: jdbc:h2:mem:testdb;MODE=PostgreSQL;DB_CLOSE_DELAY=-1;
-    username: sa
-    password: sa
+    url: jdbc:postgresql://localhost:5432/form-flow-test
```
Comment (coltborg) [inline src/test/resources/application-test.yaml:15]: I had to [add this code](https://github.com/codeforamerica/form-flow/blob/6b54e1454f995c8198abdd74b9cefb08c59f7e80/src/test/resources/application-test.yaml#L19) to get the starter app tests passing locally.

This also doesn't seem ideal. Do we have it import from the library somehow?

Comment (bencalegari) [reply src/test/resources/application-test.yaml:15]: It's totally unrelated, just copy paste from form-flow, we could rename it to make that more clear.

File: .github/workflows/run-tests.yaml [hunk 3]
```diff
@@ -13,6 +13,16 @@ jobs:
...
+      postgres:
+        image: postgres
+        env:
+          POSTGRES_USER: form-flow-test
+          POSTGRES_PASSWORD: postgres
+          POSTGRES_DB: form-flow-test
```
Comment (bencalegari) [inline .github/workflows/run-tests.yaml:22]: @coltborg It just corresponds to whatever you put here. Maybe this should just be `starter-app-test`?

Comment (coltborg) [reply .github/workflows/run-tests.yaml]: OK, changed the name to `starter-app-test`. We'll need to communicate that folks will need to create that DB/user locally. 

Maybe give them the snippet you made from the setup script?

```
# Create form-flow-test databases and users in postgres, if they don't exist
if ! psql -lqt | cut -d \| -f 1 | grep -qw starter-app-test; then
  createdb starter-app-test
  createuser -s starter-app-test
fi
```

**Pull request merged**